java/source/create.rst

```diff
@@ -30,8 +30,128 @@ Array of Int (32-bit integer value)
    intVector.set(2, 3);
    intVector.setValueCount(3);
 
-   System.out.println(intVector);
+   System.out.print(intVector);
 
 .. testoutput::
 
-    [1, 2, 3]
\ No newline at end of file
+    [1, 2, 3]
+
+
+Array of Varchar
+----------------
+
+.. testcode::
+
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.memory.RootAllocator;
+
+    RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE);
+
+    VarCharVector varVector = new VarCharVector("varVector", rootAllocator);
+    varVector.allocateNew(3);
+    varVector.set(0, "one".getBytes());
+    varVector.set(1, "two".getBytes());
+    varVector.set(2, "three".getBytes());
+    varVector.setValueCount(3);
+
+    System.out.print(varVector);
+
+.. testoutput::
+
+    [one, two, three]
+
+Array of List
+-------------
+
+.. testcode::
+
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.complex.impl.UnionListWriter;
+    import org.apache.arrow.vector.types.Types.MinorType;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.complex.ListVector;
+
+    RootAllocator rootAllocator = new RootAllocator(Long.MAX_VALUE);
+
+    ListVector listVector = ListVector.empty("listVector", rootAllocator);
+    listVector.allocateNew();
+    MinorType type = MinorType.INT;
+    listVector.addOrGetVector(FieldType.nullable(type.getType()));
+    UnionListWriter listWriter = listVector.getWriter();
+    listWriter.allocate();
```

Comment:
Is this allocate necessary? It doesn't seem needed in the examples we currently have in docs: https://arrow.apache.org/docs/java/vector.html#building-listvector

Reply:
Deleted